libpod/container_exec.go

```diff
@@ -341,22 +341,60 @@ func (c *Container) ExecStartAndAttach(sessionID string, streams *define.AttachS
 	}
 	lastErr = tmpErr
 
-	exitCode, err := c.readExecExitCode(session.ID())
-	if err != nil {
+	exitCode, exitCodeErr := c.readExecExitCode(session.ID())
+
+	// Lock again.
+	// Important: we must lock and sync *before* the above error is handled.
+	// We need info from the database to handle the error.
+	if !c.batched {
+		c.lock.Lock()
```

Comment:
I miss an `Unlock()` somewhere.

Reply:
It's way, way above

Reply:
It lives at https://github.com/containers/podman/blob/5b2597d5232627b5560bf628276ec67b35aba4bf/libpod/container_exec.go#L287

We don't use a locked/unlocked bool because there are no returns while the container is unlocked.